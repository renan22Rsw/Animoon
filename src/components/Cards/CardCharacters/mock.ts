interface mockTypes {
  id: number;
  name: string;
  image: string;
  role: string;
  voices: {
    id: number;
    name: string;
    image: string;
    language: string;
  }[];
}

export const mock: mockTypes = {
  id: 1,
  name: "Naruto",
  image: "https://example.com/character-image.jpg",
  role: "Main",
  voices: [
    {
      id: 1,
      name: "Junko Takeuch",
      image: "https://example.com/voices-image.jpg",
      language: "Japanease",
    },
  ],
};
