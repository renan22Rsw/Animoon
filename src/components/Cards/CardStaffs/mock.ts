interface mockTypes {
  id: number;
  name: string;
  image: string;
  occupation: string;
}

export const mock: mockTypes = {
  id: 1,
  name: "Masashi Kishimoto",
  image: "https://example.com/staff-image.jpg",
  occupation: "Original Creator",
};
