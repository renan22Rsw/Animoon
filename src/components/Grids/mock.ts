interface mockAnimesAndMangas {
  id: number;
  title: {
    romaji: string;
  };
  coverImage: {
    large: string;
  };
}

interface mockCharacter {
  id: number;
  name: {
    full: string;
  };
  image: {
    large: string;
  };
}

export const mockAnimesAndMangasMainPage: mockAnimesAndMangas = {
  id: 1,
  title: { romaji: "Akame ga kill" },
  coverImage: { large: "https://example.com/anime-image.jpg" },
};

export const mockCharacterMainPage: mockCharacter = {
  id: 2,
  name: { full: "Akame" },
  image: { large: "https://example.com/character-image.jpg" },
};
