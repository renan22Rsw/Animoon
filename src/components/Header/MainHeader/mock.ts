interface mockTypes {
  title: string;
  description: string;
  coverImage: string;
}

export const mock: mockTypes = {
  title: "Naruto",
  description:
    "Naruto Uzumaki, a hyperactive and knuckle-headed ninja, lives in Konohagakure, the Hidden Leaf village. Moments prior to his birth, a huge demon known as the Kyuubi, the Nine-tailed Fox, attacked Konohagakure and wreaked havoc. In order to put an end to the Kyuubi's rampage, the leader of the village, the 4th Hokage, sacrificed his life and sealed the monstrous beast inside the newborn Naruto.",
  coverImage: "https://example.com/image.jpg",
};
