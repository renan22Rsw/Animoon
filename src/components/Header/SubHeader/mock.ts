interface mockTypes {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown: string;
  yearsActive?: number[];
  description: string;
}

export const mock: mockTypes = {
  name: "Naruto",
  nativeName: "Naruto Uzumaki",
  alternativeName: ["Nine-Tails Jinchuuriki", " Number One Unpredictable"],
  age: "17",
  image: "http://example.com/image.jpg",
  gender: "Male",
  month: 10,
  day: 10,
  bloodType: "B",
  hometown: "Saitama Prefecture, Japan",
  yearsActive: [1996],
  description: "Born in Konohagakure, a ninja village hidden in the leaves",
};
