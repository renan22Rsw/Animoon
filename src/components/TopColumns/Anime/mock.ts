interface mockTypes {
  id: number;
  title: string;
  rank: number;
  images: string;
  favorites: number;
  type: string;
  episodes: number;
  status: string;
  season: string;
  year: number;
  format: string;
}

export const mock: mockTypes = {
  id: 154587,
  title: "Sousou no Frieren",
  rank: 1,
  images: "http://example.com/anime-image.jpg",
  favorites: 27371,
  type: "TV",
  episodes: 28,
  status: "FINISHED",
  season: "FALL",
  year: 2023,
  format: "Anime",
};
