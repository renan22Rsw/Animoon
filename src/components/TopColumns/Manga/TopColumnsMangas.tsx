import Image from "next/image";
import Link from "next/link";
import React from "react";
import { FaStar } from "react-icons/fa";

interface topColumnsMangas {
  title: string;
  id?: number;
  rank: number;
  images: string;
  source: string;
  meanScore: number;
  favorites: number;
  type: string;
  status: string;
  chapters: null | number;
  volumes: null | number;
}

const TopColumnsMangas = ({
  id,
  title,
  rank,
  images,
  status,
  type,
  favorites,
  chapters,
  volumes,
  meanScore,
}: topColumnsMangas) => {
  return (
    <div className="w-[90%] mx-auto flex justify-center items-center my-4">
      <h2 className="px-8 font-bold text-xl">{rank}Â°</h2>
      <div className="bg-[#1C1C1C] w-full px-4 my-4 rounded-md flex">
        <div className="h-[80px] w-[500px] flex items-center ">
          <Image
            src={images}
            width={35}
            height={35}
            alt="Image-Column"
            style={{
              width: "auto",
              height: "auto",
            }}
          />
          <Link href={`/mangas/${id}`}>
            {" "}
            <h3 className="mx-4 pb-6 font-bold text-sm">{title}</h3>
          </Link>
        </div>

        <div className="w-full grid items-center justify-end">
          <ul className="grid grid-cols-3 text-center w-[450px] ">
            <li className="px-4 font-bold ">{meanScore + "%"}</li>
            <li className="px-4 font-bold  ">{type}</li>

            <li className="px-4 font-bold capitalize">
              {volumes ? volumes + " vol " : "not finished"}
            </li>
            <li className="px-4 font-light text-xs flex items-center justify-center ">
              {favorites?.toString()}{" "}
              <FaStar className="text-yellow-400 mx-1 text-xs" />{" "}
            </li>
            <li className="px-4 font-light text-xs ">
              {chapters ? chapters + " ch" : ""}
            </li>
            <li className="px-4 font-light text-xs ">{status}</li>
          </ul>
        </div>
      </div>
    </div>
  );
};

export default TopColumnsMangas;
