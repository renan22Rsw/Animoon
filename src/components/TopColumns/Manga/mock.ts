interface mockTypes {
  title: string;
  id?: number;
  rank: number;
  images: string;
  source: string;
  meanScore: number;
  favorites: number;
  type: string;
  status: string;
  chapters: null | number;
  volumes: null | number;
}

export const mock: mockTypes = {
  title: "Berserk",
  id: 30002,
  rank: 1,
  images: "http://example.com/anime-manga.jpg",
  source: "Manga",
  meanScore: 93,
  favorites: 32311,
  type: "Manga",
  status: "RELEASING",
  chapters: null,
  volumes: null,
};
