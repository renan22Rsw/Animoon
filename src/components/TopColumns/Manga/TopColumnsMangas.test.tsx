import { render, screen } from "@testing-library/react";
import TopColumnsMangas from "./TopColumnsMangas";
import { mock } from "./mock";

describe("MangasTopColumns", () => {
  it("should render MangaTopColumns images", () => {
    render(<TopColumnsMangas {...mock} />);
    const image = screen.getByRole("img");
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute("width", "35");
    expect(image).toHaveAttribute("height", "35");
  });

  it("should render MangaTopColumns links", () => {
    render(<TopColumnsMangas {...mock} />);
    const link = screen.getByRole("link");
    expect(link).toBeInTheDocument();
    expect(link).toHaveAttribute("href", `/mangas/${mock.id}`);
  });

  it("should render MangaTopColumns infos", () => {
    render(<TopColumnsMangas {...mock} />);
    expect(screen.getByText(mock.title)).toBeInTheDocument();
    expect(screen.getByText(mock.rank + "Â°")).toBeInTheDocument();
    expect(screen.getByText(mock.source)).toBeInTheDocument();
    expect(screen.getByText(mock.meanScore + "%")).toBeInTheDocument();
    expect(screen.getByText(mock.favorites)).toBeInTheDocument();
    expect(screen.getByText(mock.type)).toBeInTheDocument();
    expect(screen.getByText(mock.status)).toBeInTheDocument();
  });

  it("should match a snapshot", () => {
    const { container } = render(<TopColumnsMangas {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
