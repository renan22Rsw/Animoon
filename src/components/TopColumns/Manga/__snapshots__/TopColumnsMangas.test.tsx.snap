// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MangasTopColumns should match a snapshot 1`] = `
<div>
  <div
    class="w-[90%] mx-auto flex justify-center items-center my-4"
  >
    <h2
      class="px-8 font-bold text-xl"
    >
      1
      Â°
    </h2>
    <div
      class="bg-[#1C1C1C] w-full px-4 my-4 rounded-md flex"
    >
      <div
        class="h-[80px] w-[500px] flex items-center "
      >
        <img
          alt="Image-Column"
          data-nimg="1"
          decoding="async"
          height="35"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Fexample.com%2Fanime-manga.jpg&w=96&q=75"
          srcset="/_next/image?url=http%3A%2F%2Fexample.com%2Fanime-manga.jpg&w=48&q=75 1x, /_next/image?url=http%3A%2F%2Fexample.com%2Fanime-manga.jpg&w=96&q=75 2x"
          style="color: transparent; width: auto; height: auto;"
          width="35"
        />
        <a
          href="/mangas/30002"
        >
           
          <h3
            class="mx-4 pb-6 font-bold text-sm"
          >
            Berserk
          </h3>
        </a>
      </div>
      <div
        class="w-full grid items-center justify-end"
      >
        <ul
          class="grid grid-cols-3 text-center w-[450px] "
        >
          <li
            class="px-4 font-bold "
          >
            93%
          </li>
          <li
            class="px-4 font-bold  "
          >
            Manga
          </li>
          <li
            class="px-4 font-bold capitalize"
          >
            not finished
          </li>
          <li
            class="px-4 font-light text-xs flex items-center justify-center "
          >
            32311
             
            <svg
              class="text-yellow-400 mx-1 text-xs"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              />
            </svg>
             
          </li>
          <li
            class="px-4 font-light text-xs "
          />
          <li
            class="px-4 font-light text-xs "
          >
            RELEASING
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
